examples/typescript/patch/patch-example.ts

```diff
@@ -0,0 +1,83 @@
+import {
+    CoreV1Api,
+    RequestContext,
+    ResponseContext,
+    KubeConfig,
+    createConfiguration,
+    Configuration,
+    ServerConfiguration,
+} from '../../../dist';
+import { PromiseMiddlewareWrapper } from '../../../dist/gen/middleware';
+
+// const k8s = require('@kubernetes/client-node');
+
+const kc = new KubeConfig();
+kc.loadFromDefault();
+
+const k8sApi = kc.makeApiClient(CoreV1Api);
+
+k8sApi
+    .listNamespacedPod({
+        namespace: 'default',
+    })
+    .then((res) => {
+        const patch = [
+            {
+                op: 'replace',
+                path: '/metadata/labels',
+                value: {
+                    foo: 'bar',
+                },
+            },
+        ];
+        const options = { headers: { 'Content-type': 'application/json-patch+json' } };
+        const headerPatchMiddleware = new PromiseMiddlewareWrapper({
+            pre: async (requestContext: RequestContext) => {
+                requestContext.setHeaderParam('Content-type', 'application/json-patch+json');
+                return requestContext;
+            },
+            post: async (responseContext: ResponseContext) => responseContext,
```

Comment:
Hey, @OliverMKing, one question: 
is it required to define both `pre` as well as `post` even if you do not do anything else than return the arguments like in your example with `post`?

Reply:
i [don't see any default implementations ](https://github.com/kubernetes-client/javascript/blob/ca53fef7f65f8544a2e98db7caaff2d0e92e0152/src/gen/middleware.ts#L28)of the pre/post methods, so they are most likely required for now,  but we could easily add a wrapper that does default them or even a wrapper that makes middleware/configuration from a desired MIME media type since that's most of what we're after here

Reply:
you can skip the middleware altogether if you want to use [the first supported content type](https://github.com/davidgamero/javascript/blob/c17ff3193dbd111b9e5b5db939647af70005381e/src/gen/apis/CoreV1Api.ts#L10138C10-L10138C10) which is `application/json-patch+json`, but for any other types you'll have to add it